packages/vitest/src/node/stdin.ts

```diff
@@ -119,15 +119,49 @@ export function registerConsoleShortcuts(ctx: Vitest) {
 
   async function inputFilePattern() {
     off()
-    const { filter = '' }: { filter: string } = await prompt([{
+    // implement same filtering as ctx.rerunFiles
+    const allFiles = ctx.state.getFilepaths()
+    const allFilesSet = new Set(allFiles)
+    async function filterFiles(input: string) {
+      const globTestFiles = await ctx.globTestFiles([input])
+      return globTestFiles.map(spec => spec[1]).filter(v => allFilesSet.has(v))
```

Comment:
The other PR implemented this correctly - it filters all available test files. I am not sure what happens here yet, but it only filters test files that were already filtered (so, with `vitest filter` this only suggests files with filter in the name, and not all available test files)